src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.RegionBuilder.cs

```diff
@@ -77,19 +77,82 @@ public void Add(RegionBuilder region)
                         }
                         else
                         {
-                            Debug.Assert(lastKind == ControlFlowGraph.RegionKind.Handler);
+                            Debug.Assert(lastKind == ControlFlowGraph.RegionKind.Finally);
                         }
                         break;
 
                     default:
                         Debug.Assert(lastKind != ControlFlowGraph.RegionKind.Filter);
-                        Debug.Assert(lastKind != ControlFlowGraph.RegionKind.Handler);
+                        Debug.Assert(lastKind != ControlFlowGraph.RegionKind.Catch);
+                        Debug.Assert(lastKind != ControlFlowGraph.RegionKind.Finally);
                         Debug.Assert(lastKind != ControlFlowGraph.RegionKind.Try);
                         break;
                 }
 #endif
             }
 
+            public void Remove(RegionBuilder region)
+            {
+                Debug.Assert(region.Enclosing == this);
+
+                if (Regions.Count == 1)
```

Comment:
Consider removing the special cases for `Regions.Count == 1` (here and in `ReplaceRegion`) unless the behavior is different from `> 1` case.

Reply:
> Consider removing the special cases for Regions.Count == 1 (here and in ReplaceRegion) unless the behavior is different from > 1 case. 

I think this optimization is very simple and worth keeping. The code path is covered by unit-tests.

---
In reply to: [171017423](https://github.com/dotnet/roslyn/pull/25079#discussion_r171017423) [](ancestors = 171017423)